### PR TITLE
feat(inputs): update to latest styles

### PR DESCRIPTION
**Pull request by sandgraham**
#### Description

Updates `Input`, `Select`, and `Menu` styles to match latest Base designs:

- Update rounded corners to square corners
- Update to thicker border width
- Update "active" (focus) state colors (active is now `black`)
- Update "negative" (error) state colors (lighter `red`)
- Update "disabled" state colors (slightly different `grey`s)
- Add positive state
- Add large sizes
- Enhancers now take on state of input (generally take on border color)

#### Existing inputs:

Screen Shot 2019-05-08 at 5 21 45 PM
Screen Shot 2019-05-08 at 5 22 31 PM

#### Existing select & menu:
Screen Shot 2019-05-10 at 10 20 08 AM

#### New inputs:

Screen Shot 2019-05-10 at 10 21 47 AM

#### Updated selects & menu:

Screen Shot 2019-05-10 at 10 18 52 AM
Screen Shot 2019-05-10 at 10 19 06 AM

#### Updated sizing for inputs (new large size):

Screen Shot 2019-05-10 at 10 23 09 AM

#### Scope

- [ ] Patch: Bug Fix
- [ ] Minor: New Feature
- [x] Major: Breaking Change

### CODE REVIEW

**Comment by gergelyke** [comment]
this looks great! should we include the select component in this change too? if the input and the select are used in the same form, they look odd next to each other

**Comment by sandgraham** [comment]
@gergelyke Yeah they seem like an odd pair. Let me update that too- need to get some design input since I don't think those have been mocked up yet.

File: src/input/styled-components.js [hunk 1]
```diff
@@ -62,96 +30,232 @@ export const Root = styled('div', props => {
...
+      paddingRight: sizing.scale400,
+      paddingLeft: sizing.scale400,
+    },
+    [SIZE.default]: {
+      paddingRight: sizing.scale500,
+      paddingLeft: sizing.scale500,
+    },
+    [SIZE.large]: {
+      paddingRight: sizing.scale600,
+      paddingLeft: sizing.scale600,
```
Comment (sandgraham) [inline src/input/styled-components.js]: @nadiia is something like this okay?

Comment (nadiia) [reply src/input/styled-components.js]: Yes, perfect!

**Comment by gergelyke** [comment]
it seems like this PR changes the side-navigation too (the elements are a lot higher) - is that intentional?

**Comment by sandgraham** [comment]
I'm not sure why that has changed.
Here is where the background color is controlled:
```js
// src/side-navigation/styled-components.js
// L40
backgroundColor: $active
      ? `linear-gradient(0deg, rgba(255, 255, 255, 0.92), rgba(255, 255, 255, 0.92)), ${
          colors.primary
        }`
      : 'transparent',
```
I haven't changed `primary` or the `$active` prop so far as I know... 🤔
Will have to dig into this a bit.

**Comment by sandgraham** [comment]
Looks like that was introduced here: https://github.com/uber-web/baseui/commit/c5fe7472e996c38ba77bd2f459b027f2e190ce0f#diff-97acb7bcb62a2cc242b1860993b8425fL44

`linear-gradient` can only be used where `<image>` can be used (in CSS). So the property needs to be `background` or `background-image`. In this case we are using 2 background layers (one to lighten our primary blue) and so we either use short hand`background` to apply both or split into `background-color` and `background-image` for the gradient if we require long hand forms.

**Comment by gergelyke** [comment]
it's strange, as on the current master, it looks good 🤔

**Comment by sandgraham** [comment]
oh i rebased onto `v7` - i included a quick fix in this branch

File: src/input/types.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ export type SharedPropsT = {
   $disabled: boolean,
   /** Renders UI in 'error' state */
   $error: boolean,
+  /** Renders UI in 'positive' state */
+  $positive: boolean,
```
Comment (tajo) [inline src/input/types.js]: what happens if I use both `$positive` and `$negative` at the same time?

Comment (tajo) [reply src/input/types.js]: it seems that `$error` wins, a warning could be nice?

Comment (sandgraham) [reply src/input/types.js]: Good point, I'll add the warning. Might be worth using a `STATE` enum or something if these get more complicated.

**Pull request merged**